tests/end_to_end/test_runtimes_e2e.py

```diff
@@ -0,0 +1,106 @@
+import json
+from unittest import TestCase
+
+import os
+
+from pathlib import Path
+
+from parameterized import parameterized_class
+
+from tests.end_to_end.end_to_end_base import EndToEndBase, EndToEndTestContext
+from tests.end_to_end.test_stages import (
+    DefaultInitStage,
+    DefaultRemoteInvokeStage,
+    DefaultDeleteStage,
+    EndToEndBaseStage,
+    DefaultSyncStage,
+)
+from tests.testing_utils import CommandResult
+
+
+class HelloWorldValidators(TestCase):
+    def default_command_output_validator(self, command_result: CommandResult):
+        self.assertEqual(command_result.process.returncode, 0)
+
+    def validate_init(self, command_result: CommandResult):
```

Comment:
I wonder if we should split them up in to their own classes.

Example: InitValidator, BuildValidator. This way the method name could be validate and it will be consistent across all validators. And we have a base level validator which always checks for return code.

Reply:
It also maps with the stage validator concept that you have already.